libevmasm/AssemblyItem.cpp

```diff
@@ -207,11 +209,21 @@ string AssemblyItem::toAssemblyText() const
 		text = string("data_") + util::toHex(data());
 		break;
 	case PushSub:
-		text = string("dataOffset(sub_") + to_string(static_cast<size_t>(data())) + ")";
+	{
+		vector<string> subIdsString;
+		for (size_t subId: Assembly::decodeSubIds(data()))
+			subIdsString.emplace_back("sub_" + to_string(subId));
+		text = string("dataOffset(") + solidity::util::joinHumanReadable(subIdsString, ".") + ")";
```

Comment:
Not sure if it's all that much better, but you could also do:
```suggestion
		text = string("dataOffset(") +
			solidity::util::joinHumanReadable(Assembly::decodeSubIds(data()) | boost::adaptors::transformed(to_string), ".") +
			")";
```